src/main/java/bio/terra/cli/service/WorkspaceManager.java

```diff
@@ -228,89 +245,224 @@ public RoleBindingList listUsersOfWorkspace() {
+                    name,
+                    description,
+                    cloningInstructions,
+                    accessScope,
+                    gcsBucketName,
+                    defaultStorageClass,
+                    lifecycleRules,
+                    location));
+  }
+
+  /**
+   * Add a Big Query dataset as a controlled resource in the workspace. Also updates the cached list
+   * of resources.
+   *
+   * @param name name of the resource. this is unique across all resources in the workspace
+   * @param description description of the resource
+   * @param cloningInstructions instructions for how to handle the resource when cloning the
+   *     workspace
+   * @param accessScope access to allow other workspaces users
+   * @param bigQueryDatasetId Big Query dataset id
+   *     (https://cloud.google.com/bigquery/docs/datasets#dataset-naming)
+   * @param location Big Query dataset location (https://cloud.google.com/bigquery/docs/locations)
+   * @return the resource description object that was created
+   */
+  public ResourceDescription createControlledBigQueryDataset(
+      String name,
+      String description,
+      CloningInstructionsEnum cloningInstructions,
+      AccessScope accessScope,
+      String bigQueryDatasetId,
+      @Nullable String location) {
+    return createResource(
+        name,
+        () ->
+            new WorkspaceManagerService(
+                    globalContext.server, globalContext.requireCurrentTerraUser())
+                .createControlledBigQueryDataset(
+                    workspaceContext.getWorkspaceId(),
+                    name,
+                    description,
+                    cloningInstructions,
+                    accessScope,
+                    bigQueryDatasetId,
+                    location));
+  }
+
+  /**
+   * Create a new resource in the workspace. Also updates the cached list of resources.
    *
-   * @param resourceType type of resource to create
    * @param resourceName name of resource to create
-   * @return the cloud resource that was created
+   * @param createResource function pointer to execute the create request for a particular
+   *     stewardship + resource type combination
+   * @param <T> type of the create response object. this is usually very similar to the
+   *     ResourceDescription object, but not quite identical.
+   * @return the resource description object that was created
    */
-  public CloudResource createControlledResource(
-      CloudResource.Type resourceType, String resourceName) {
+  private <T> ResourceDescription createResource(String resourceName, Supplier<T> createResource) {
     workspaceContext.requireCurrentWorkspace();
-    // TODO: change this method to call WSM controlled resource endpoints once they're ready
+
     if (!isValidEnvironmentVariableName(resourceName)) {
       throw new UserActionableException(
           "Resource name can contain only alphanumeric and underscore characters.");
```

Comment:
do we still need to do this validation or is it ok to rely on WM? I think we'll get out of date on what's allowed, and it will be possible to create resources from paths other than the CLI.

Reply:
At the moment, we still need this validation because WSM is not enforcing it. The last time I brought this up, I think we decided to punt on making a decision.

Another approach would be to stop passing environment variables into the container, install the Terra CLI in the container, and then force apps to call `terra resources resolve ...` instead of relying on environment variables. This would also remove the need to escape things.

Regardless, I think this should be addressed in a separate PR, after some discussion and possibly prototyping/demoing with the solutions team code.